### PR TITLE
#154 - Implement unpublishing single version, add IT

### PR DESCRIPTION
**Pull request by genryxy**
Part of #154
Unpublishing single version was implemented. 
Added IT for removing single version when multiple versions are published and one version is published

### CODE REVIEW

File: src/main/java/com/artipie/npm/http/UnpublishPutSlice.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    }
+
+    /**
+     * Compare two meta files and remove from the meta file of storage info about
+     * version that does not exist in another meta file.
+     * @param body Meta json file (usually this file is received from body)
+     * @param meta Meta json from storage
+     * @return Meta json with removed information about unpublished version.
+     */
+    private static String updateMeta(final JsonObject body, final JsonObject meta) {
```
Comment (g4s8) [inline src/main/java/com/artipie/npm/http/UnpublishPutSlice.java:117]: @genryxy param names (`body` and `meta`) looks not very clean - both are the same kind of data but from different sources. I'd rename it to something like `patch` or `update` for new meta json and `source` or `origin` for origin json

File: src/main/java/com/artipie/npm/http/UnpublishPutSlice.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        return patch.build().apply(meta).toString();
+    }
+
+    /**
+     * Compare two meta files and identify which version does not exist in one of meta files.
+     * @param body Meta json file (usually this file is received from body)
+     * @param meta Meta json from storage
+     * @return Version to unpublish.
+     */
+    private static String versionToRemove(final JsonObject body, final JsonObject meta) {
```
Comment (g4s8) [inline src/main/java/com/artipie/npm/http/UnpublishPutSlice.java:136]: @genryxy and same here

**Review by g4s8 — CHANGES_REQUESTED** [review]
@genryxy please check my comments

**Comment by genryxy** [comment]
@g4s8 thanks, renamed

**Review by g4s8 — APPROVED** [review]
@genryxy thanks

**Comment by 0crat** [comment]
Job `gh:artipie/npm-adapter#189` is not assigned, can't get performer

**Pull request merged**